docs/proposals/migration/design-of-seamless-cluster-migration-scheme.md

```diff
@@ -0,0 +1,291 @@
+---
+title: proposal of seamless cluster migration
+authors:
+- "@chaosi-zju"
+reviewers:
+- "@RainbowMango"
+- TBD
+approvers:
+- "@RainbowMango"
+- TBD
+
+creation-date: 2023-07-08
+---
+
+# Proposal of seamless cluster migration
+
+
+
+## Summary
+
+- When users migrate a single cluster to multiple clusters, there is a common troublesome problem: how to handle the resources that already exist in the original cluster? In Most scenario, users hope that `Karmada` can maintain the previous running state of existing resources in the original cluster during the process of taking over the original cluster. We refer to this migration method as seamless migration.
+
+- Currently, there are several ways for `Karmada` to achieve seamless migration like [karmadactl promote](https://karmada.io/docs/administrator/migration/promote-legacy-workload), but the existing methods require operators to perceive the resources of the original cluster, require user intervention, and the user experience is not smooth enough.
+
+- This proposal aims to further optimizing of seamless migration by extending the `Karmada` custom resource API, ensuring that users are completely unaware during the migration process.
+
+  
+
+
+## Motivation
+
+Users migrate a single cluster to multiple clusters, hoping that `Karmada` will take over the existing deployment or other resources of the original single cluster. During the migration process, users hope to achieve seamless switching, including:
+
+- **Pods that already exist during the entire migration process will not be restarted**.
+
+- When deploying Pod to a new cluster, there will be a verification process, and the number of Pod replicas in the original cluster will remain unchanged until the verification is completed.
+
+- After the migration is completed, the total number of Pod replicas in all clusters remains unchanged.
+
+
+
+Karmada currently has two ways to achieve seamless migration as mentioned above:
+
+- **Method 1:** Call `karmadactl promote deployment xxx -C yyy` to take over the existing deployment and other resources in the original single cluster
+- **Method 2:** Manually write the `PropagationPolicy` and `ResourceTemplate` configurations and call the ApiServer interface of the Host cluster to apply the configuration. Noting that **it is necessary to add `work.karmada.io/conflict-resolution: overwrite` in the annotation of `ResourceTemplate`**, indicating that when there is a naming conflict resource in the member cluster, it will be forced taken over by overwriting the configuration **(if left blank, it will not be taken over by default)** 
+
+
+> Detailed practical steps can be found in: [Utilizing existing policy capabilities to achieve cluster migration](https://www.yuque.com/chaosi/pwzt9c/ypugrp6452lu36ok)
```

Comment:
The link is dead.
In addition, please try to link documents from the Karmada community or a stable website.

Reply:
directly removed it